src/arch/arm/64/machine/capdl.c

```diff
@@ -395,12 +397,12 @@ void obj_frame_print_attrs(vm_page_size_t frameSize, paddr_t frameBase)
         break;
     }
 
-    printf(", paddr: 0x%p)\n", (void *)frameBase);
+    printf(", paddr: 0x%lx)\n", frameBase);
 }
 
 void arm64_obj_pt_print_slots(pte_t *pdSlot)
 {
-    pte_t *pt = paddr_to_pptr(pte_pte_table_ptr_get_pt_base_address(pdSlot));
+    pte_t *pt = PT_PTR(paddr_to_pptr(pte_pte_table_ptr_get_pt_base_address(pdSlot)));
```

Comment:
paddr_to_pptr / ptrFromPAddr return `void*`, why casting that? You seem to make a lot of unnecessary changes, but you must have a good reason.